### PR TITLE
Added `BookshelfMilestoneRepository` implementation

### PR DESCRIPTION
**Pull request by aileen**
needs https://github.com/TryGhost/Ghost/pull/16300
refs https://www.notion.so/ghost/Marketing-Milestone-email-campaigns-1d2c9dee3cfa4029863edb16092ad5c4?pvs=4

This stores the received milestones in the database.

### CODE REVIEW

File: ghost/core/test/e2e-server/services/milestones.test.js [hunk 1]
```diff
@@ -156,7 +157,8 @@ describe('Milestones Service', function () {
     beforeEach(async function () {
         loggingStub = sinon.stub(logging, 'info');
         const now = new Date();
-        clock = sinon.useFakeTimers(now.getTime());
+        const threeMonthsAgo = now.setMonth(now.getMonth() - 3);
```
Comment (allouis) [inline ghost/core/test/e2e-server/services/milestones.test.js:160]: The `setMonth` method mutates the `Date` instance, so after this line the `now` variable is actually three months ago. Obviously fine here but if someones changes the tests and tries to use the `now` value they'll get unexpected results!

Also the value of `new Date(threeMonthsAgo).getTime()` and `threeMonthsAgo` are the same! Date objects are freaking weird.

I would probably change this to something like:
```js
const threeMonthsAgo = new Date();
threeMonthsAgo.setMonth(threeMonthsAgo.getMonth() - 3);
clock = sinon.useFakeTimers(threeMonthsAgo.getTime());
```

File: ghost/core/test/e2e-server/services/milestones.test.js [hunk 2]
```diff
@@ -11,6 +11,7 @@ const milestonesService = require('../../../core/server/services/milestones');
 let agent;
 let counter = 0;
 let membersCounter = 0;
+const inFifteenDays = 1000 * 60 * 60 * 24 * 15;
```
Comment (allouis) [inline ghost/core/test/e2e-server/services/milestones.test.js:14]: I don't really care, but we would generally call this `fifteenDays` without the `'in'` prefix

**Review by allouis — APPROVED** [review]
The repository looks good - I'm unsure why the tests have changed though they should all be independent of any storage concerns 🤔 

If possible I would try and separate out these changes into separate commits/PR - if they can't be separated I'd be interested to know what we need to change the tests, it might be a sign we're too tightly coupled

File: ghost/core/test/e2e-server/services/milestones.test.js (hunk 1 above)
Comment (aileen) [reply ghost/core/test/e2e-server/services/milestones.test.js]: Moved test updates to https://github.com/TryGhost/Ghost/pull/16314

File: ghost/core/test/e2e-server/services/milestones.test.js (hunk 2 above)
Comment (aileen) [reply ghost/core/test/e2e-server/services/milestones.test.js]: Moved test updates to https://github.com/TryGhost/Ghost/pull/16314

**Pull request merged**